projects/packages/sync/src/modules/class-woocommerce-products.php

```diff
@@ -353,4 +360,171 @@ public function get_next_chunk( $config, $status, $chunk_size ) {
 			'objects'    => $filtered_product_data,
 		);
 	}
+
+	/**
+	 * Get the product meta data from the product meta lookup table.
+	 *
+	 * @param array  $ids List of product IDs to fetch.
+	 * @param string $order Either 'ASC' or 'DESC'.
+	 *
+	 * @return array
+	 */
+	private function get_product_meta_data( $ids, $order = '' ) {
+		global $wpdb;
+
+		// Prepare the placeholders for the prepared query below.
+		$placeholders = implode( ',', array_fill( 0, count( $ids ), '%d' ) );
+
+		$query = "SELECT * FROM {$this->table()} WHERE product_id IN ( $placeholders )";
+		if ( ! empty( $order ) && in_array( $order, array( 'ASC', 'DESC' ), true ) ) {
+			$query .= " ORDER BY product_id $order";
+		}
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Hardcoded query, no user variable
+		$product_meta_data = $wpdb->get_results( $wpdb->prepare( $query, $ids ), ARRAY_A );
+
+		if ( ! is_array( $product_meta_data ) ) {
+			return array();
+		}
+
+		return $product_meta_data;
+	}
+
+	/**
+	 * Get the product data from the posts table.
+	 *
+	 * @param array  $ids List of product IDs to fetch.
+	 * @param string $order Either 'ASC' or 'DESC'.
+	 *
+	 * @return array
+	 */
+	private function get_product_posts( $ids, $order = '' ) {
+		$posts = get_posts(
+			array(
+				'include'     => $ids,
+				'order'       => $order,
+				'post_type'   => array( 'product', 'product_variation' ),
+				'post_status' => 'any',
+				'numberposts' => -1, // Get all posts.
```

Comment:
[There could be very large stores](https://pressable.com/blog/scaling-woocommerce-for-large-stores-handling-100000-products/) with 100k+ products. I think loading all posts at once could cause performance issues in such cases. Can we paginate with a safe max number of posts per page, like 100 or something?